### PR TITLE
Add a datepicker

### LINKED ISSUE #3: Introduce a DateSelector
Right now I can write down pretty much everything in the field for Dates.
Introducing a DateSelector would be beneficial for this application, because it wouldn't allow consumers to write down nonsense in the date field. 
You can use any DateSelector library you like. 
Make sure that it is **used by many developers** and that **it is maintained**.

### PR DESCRIPTION
**Pull request by TheaAleksovDev**

### CODE REVIEW

**Comment by valentin-aleksandrov** [comment]
Please, use present simple for commits:
image

Since, you have already have used git merge, rebasing this branch wouldn't be possible without a higher level of understanding git.
Let's keep this commit like this, but keep the above suggestion in mind for future PRs.

File: src/app/components/add-note-popup/add-note-popup.component.ts [hunk 1]
```diff
@@ -37,9 +49,9 @@ export class AddNotePopupComponent {
     this.addedNote.emit(this.formData.subject);
   }
 
-  private convertToDate(dateString: string): Date {
-    const [day, month, year] = dateString.split('-').map(Number);
-    return new Date(year, month - 1, day);
+  onDateChange(event: any): void {
+    console.log(event.target.value);
```
Comment (valentin-aleksandrov) [inline src/app/components/add-note-popup/add-note-popup.component.ts]: **Not important**:
```suggestion
```

File: src/app/components/add-note-popup/add-note-popup.component.ts [hunk 2]
```diff
@@ -37,9 +49,9 @@ export class AddNotePopupComponent {
     this.addedNote.emit(this.formData.subject);
   }
 
-  private convertToDate(dateString: string): Date {
-    const [day, month, year] = dateString.split('-').map(Number);
-    return new Date(year, month - 1, day);
+  onDateChange(event: any): void {
```
Comment (valentin-aleksandrov) [inline src/app/components/add-note-popup/add-note-popup.component.ts]: Can you type the event?

File: src/app/components/add-note-popup/add-note-popup.component.ts
Comment (valentin-aleksandrov) [inline src/app/components/add-note-popup/add-note-popup.component.ts:1]: image

File: src/app/components/note/note.component.html [hunk 3]
```diff
@@ -19,7 +19,9 @@ <h2 class="note-title">
         >delete</mat-icon
       >
     </h2>
-    <h4 class="note-date" *ngIf="selectedNote === note.id">{{ note.date }}</h4>
+    <h4 class="note-date" *ngIf="selectedNote === note.id">
+      {{ note.date | date }}
```
Comment (valentin-aleksandrov) [inline src/app/components/note/note.component.html:23]: Does the `date` variable exists?
fox-mulder-xfiles

Comment (TheaAleksovDev) [reply src/app/components/note/note.component.html:23]: I'm using date from DatePipe, actually

Comment (valentin-aleksandrov) [reply src/app/components/note/note.component.html:23]: Let's say that's the case.
Should the following code change still works:
image

Comment (valentin-aleksandrov) [reply src/app/components/note/note.component.html:23]: You're right. I confused it with something, which looks similar in React, but does entirely different job.

**Review by valentin-aleksandrov — APPROVED** [review]

**Pull request merged**